src/targeting/Find_targets.ml

```diff
@@ -452,12 +473,33 @@ let get_targets_for_project conf (project_roots : project_roots) =
   in
   (selected_targets, skipped_targets)
 
+let setup_project_roots conf scanning_roots =
+  match conf.project_root with
+  | Some (Filesystem _) -> scanning_roots
+  | Some (Git_remote { url; checkout_path }) ->
+      Logs.debug (fun m ->
+          m "Sparse cloning %a into %a" Uri.pp url Fpath.pp checkout_path);
+      (match Git_wrapper.sparse_shallow_filtered_checkout url checkout_path with
+      | Ok () -> ()
+      | Error msg ->
+          Logs.err (fun m ->
+              m "Error while sparse cloning %a into %a: %s" Uri.pp url Fpath.pp
+                checkout_path msg);
+          exit 1);
```

Comment:
FYI @ajbt200128 you can't use exit. There's actually now a semgrep check for it, but this file had an issue with tree-sitter-ocaml (that I fixed), so semgrep didn't report it. First if you use exit, you need to use special exit code for semgrep (see Exit_code.ml), and now you also need the exit capability. I'll fix the code, but just FYI.

Reply:
Thanks, I meant to switch it to error.abort or - [ ] something similar but missed it